Tasks/AzureRmWebAppDeployment/task.json

```diff
@@ -76,19 +81,47 @@
             "visibleRule": "DeployToSlotFlag = true"
         },
         {
+            "name": "VirtualApplication",
+            "type": "string",
+            "label": "Virtual Application",
+            "defaultValue": "",
+            "required": false,
+            "helpMarkDown": "Specify the name of the Virtual Application that has been configured in the Azure portal. The option is not required for deployments to the website root."
+        },
+        {
             "name": "Package",
             "type": "filePath",
-            "label": "Package",
+            "label": "Package or Folder",
             "defaultValue": "",
             "required": true,
             "helpMarkDown": "Location of the Web App zip package on the automation agent or on a UNC path accessible to the automation agent like, \\\\\\\\BudgetIT\\Web\\Deploy\\Fabrikam.zip. Predefined system variables and wild cards like, $(System.DefaultWorkingDirectory)\\\\***.zip` can be also used here."
         },
         {
+            "name": "WebAppUri",
+            "type": "string",
+            "label": "Web App Hosted Url",
+            "required": false,
+            "defaultValue": "",
+            "groupName": "output",
+            "helpMarkDown": "Provide a name for the variable for the webapp hosted url. The variable can be used as $(variableName) to refer to the hosted url of webapp in subsequent tasks like in the PowerShell on Target Machines task."
```

Comment:
Specify a name for the output variable that is generated for the URL of the Web App. The variable can be consumed in subsequent tasks.